Framework/Algorithms/test/FindPeakBackgroundTest.h

```diff
@@ -184,51 +180,22 @@ class FindPeakBackgroundTest : public CxxTest::TestSuite {
   /** Generate a workspace with 2 spectra for test
    */
   MatrixWorkspace_sptr generate2SpectraTestWorkspace() {
-    vector<double> data;
-    data.push_back(1);
-    data.push_back(2);
-    data.push_back(1);
-    data.push_back(1);
-    data.push_back(9);
-    data.push_back(11);
-    data.push_back(13);
-    data.push_back(20);
-    data.push_back(24);
-    data.push_back(32);
-    data.push_back(28);
-    data.push_back(48);
-    data.push_back(42);
-    data.push_back(77);
-    data.push_back(67);
-    data.push_back(33);
-    data.push_back(27);
-    data.push_back(20);
-    data.push_back(9);
-    data.push_back(2);
+
+    const size_t size = 20;
 
     MatrixWorkspace_sptr ws = boost::dynamic_pointer_cast<MatrixWorkspace>(
-        WorkspaceFactory::Instance().create("Workspace2D", 2, data.size(),
-                                            data.size()));
+        WorkspaceFactory::Instance().create("Workspace2D", 2, size, size));
 
     // Workspace index = 0
-    MantidVec &vecX = ws->dataX(0);
-    MantidVec &vecY = ws->dataY(0);
-    MantidVec &vecE = ws->dataE(0);
-    for (size_t i = 0; i < data.size(); ++i) {
-      vecX[i] = static_cast<double>(i);
-      vecY[i] = 0.0;
-      vecE[i] = 1.0;
-    }
+    MantidVec xdata(size);
+    std::iota(xdata.begin(), xdata.end(), 0);
+    ws->mutableX(0) = std::move(xdata);
+    ws->mutableE(0).assign(size, 1.0);
 
     // Workspace index = 1
-    MantidVec &vecX1 = ws->dataX(1);
-    MantidVec &vecY1 = ws->dataY(1);
-    MantidVec &vecE1 = ws->dataE(1);
-    for (size_t i = 0; i < data.size(); ++i) {
-      vecX1[i] = static_cast<double>(i);
-      vecY1[i] = data[i];
-      vecE1[i] = sqrt(data[i]);
-    }
+    ws->setSharedX(1, ws->sharedX(0));
+    ws->setCounts(1, Counts{1, 2, 1, 1, 9, 11, 13, 20, 24, 32, 28, 48, 42, 77,
+                            67, 33, 27, 20, 9, 2});
```

Comment:
The errors are not auto-generated when you use the individual setters, otherwise any modification of `Y` would overwrite the error values!

Auto-generation happens only in the `Histogram` constructor, so when you use `setHistogram()`.